.github/workflows/compile-wasm.yml

```diff
@@ -12,15 +12,13 @@ jobs:
     steps:
       - uses: actions/checkout@v4
       - id: compiled
-        uses: actions/cache@v3 # turbo cache, shared between buildjet and github
+        uses: buildjet/cache@v3
         with:
           path: .turbo
-          key: ${{ runner.os }}-turbo-${{ hashFiles('**/Cargo.lock') }}-${{ hashFiles('pnpm-lock.yaml') }}
-          restore-keys: |
-            ${{ runner.os }}-turbo-${{ hashFiles('**/Cargo.lock') }}
-            ${{ runner.os }}-turbo-
+          key: ${{ hashFiles('**/Cargo.lock') }}-${{ hashFiles('pnpm-lock.yaml') }}-${{ github.ref }}-${{ github.sha }}-compiled
```

Comment:
Hmm, a couple notes:
1. [`hashFiles` can take multiple arguments](https://docs.github.com/en/actions/learn-github-actions/expressions#example-with-multiple-patterns), so you could rewrite the first part of the key as: `${{ hashFiles('**/Cargo.lock', 'pnpm-lock.yaml') }}`
2. But more importantly: Wouldn't it actually be better to cache based on the hash of all files in the `packages/wasm/crate` directory, so that we only compile WASM when any WASM files change? (And I'd remove the `github.ref` and `github.sha`, since those can change without the WASM code changing. Right?)

Reply:
i had trouble with getting `hashFiles` to work on multiple files, with directories, or with globs. it successfully returns an empty string on any failure, and i can't test these actions locally on my m3.

i agree such a change would be an improvement but i don't want to spend more time on this. PR it if you'd like